plone/app/iterate/relation.py

```diff
@@ -62,21 +68,29 @@ class CheckinCheckoutReferenceAdapter ( object ):
 
     forward refs on baseline get removed
     backward refs on baseline are kept by virtue of UID transferance
-    
+
     """
 
     implements( ICheckinCheckoutReference )
     adapts( IIterateAware )
-    
+
     storage_key = "coci.references"
 
     def __init__(self, context ):
         self.context = context
-    
-    def checkout( self, baseline, wc, refs, storage ):
+
+    def checkout(self, baseline, wc, refs, storage):
         for ref in refs:
-            wc.addReference( ref.targetUID, ref.relationship, referenceClass=ref.__class__ )
-            
+            try:
+                wc.addReference(ref.targetUID, ref.relationship,
+                                referenceClass=ref.__class__)
+            except ReferenceException:
+                logger.warn(
+                    'Reference exception when adding relation %r '
+                    'from new working copy %s to uid %s. Ignoring relation.',
+                    ref.relationship, '/'.join(wc.getPhysicalPath()),
+                    ref.targetUID)
+
```

Comment:
This part of the code takes the list of references from the original and for each one adds a reference in the working copy. When the `addReference` call fails for a reference, it does not get added.

I guess instead of 'remove broken references when making checkout' I could have called this pull request 'don't copy broken references when making checkout'. :-)